### DISCUSSION TITLE
How to change rust-analyzer cfg directives?

### DISCUSSION THREAD

**Discussion by damccull** [mrcjkb/rustaceanvim#484] [Q&A]
When running this code in my nvim setup, rust-analyzer is graying out code for cfg directives that it doesn't have enabled. In vscode I was able to set a setting in the workspace settings file that would change the enable feature flags for rust-analyzer, but I would have to restart vscode (or at least rust-analyzer) for it to switch. This was useful because I wouldn't get auto-complete and similar things on non-enabled code.

How do I do this with rustacean-vim, preferably without changing a settings file directly. Even if I have to restart the LSP, I'd like to be able to set the setting live in the editor to swap back and forth easily.

**Comment by mrcjkb** [comment]
> How do I do this with rustacean-vim, preferably without changing a settings file directly. Even if I have to restart the LSP, I'd like to be able to set the setting live in the editor to swap back and forth easily.

Neovim, and by extension rustaceanvim, has no such feature. You have to edit your Neovim config or your .vscode/settings.json manually.

For configuring rustaceanvim and rust-analyzer, see `:h rustaceanvim.config`.
